### DISCUSSION TITLE
dimensions of electrode_index in Units table

### DISCUSSION THREAD

**Discussion by magland** [dandi/helpdesk#131] [General]
I don't know if this is too minor to report, but I noticed in at least one dataset the shape of electrode_index in the Units table is `(x, 1)` rather than just `(x,)`. According to the spec (https://nwb-schema.readthedocs.io/en/latest/format.html#units) this should be 1-d, not 2-d, I believe. This came up because neurosift failed to display that column. I'll update neurosift to be able to handle this case, but wanted to draw it to your attention. Is this something that should have been detected during NWB validation?

Here's the file
https://neurosift.app/?p=/nwb&dandisetId=000939&dandisetVersion=0.240327.2229&url=https://api.dandiarchive.org/api/assets/56d875d6-a705-48d3-944c-53394a389c85/download/

Go to the raw tab, then expand units/electrode_index. You'll see that the shape is `[101, 1]`.

**Comment by rly** [comment]
@magland The `electrodes_index` column of the `Units` table has neurodata type `VectorIndex` which must indeed be 1-d.

However, the file you linked contains a custom column called `electrode_index` (without the "s"). This has neurodata type `VectorData` which can be multidimensional. It's a bit silly to make the dataset have shape `(x, 1)` instead of `(x, )` but it is valid. 

Did you come across a different file with `Units/electrodes_index` with the "s" that has the issue?

**Reply by magland** [reply]
Thanks Ryan, I didn't notice the "s" there.
